perun/postprocess/regression_analysis/methods.py

```diff
@@ -0,0 +1,445 @@
+"""Module with available regression computational methods.
+
+This module exposes all currently implemented computational methods for regression analysis.
+
+"""
+
+import collections
+
+import perun.postprocess.regression_analysis.regression_models as mod
+import perun.utils.exceptions as exceptions
+import perun.postprocess.regression_analysis.tools as tools
+
+
+def get_supported_methods():
+    """Provides all currently supported computational methods as a list of their names.
+
+    Returns:
+        list of str: the names of all supported methods
+    """
+    return [key for key in _METHODS.keys()]
+
+
+def compute(data_gen, method, models, **kwargs):
+    """The regression analysis wrapper for various computation methods.
+
+    Arguments:
+        data_gen(iter): the generator object with collected data (data provider generators)
+        method(str): the _METHODS key value indicating requested computation method
+        models(list of str): list of requested regression models to compute
+        kwargs: various additional configuration arguments for specific models
+    Raises:
+        GenericRegressionExceptionBase: derived versions which are used in the computation functions
+        DictionaryKeysValidationFailed: in case the data format dictionary is incorrect
+    Returns:
+        list of dict: the computation results
+
+    """
+    # Initialize the resulting dictionary
+    analysis = []
+
+    for chunk in data_gen:
+        try:
+            for result in _METHODS[method](chunk[0], chunk[1], models, **kwargs):
+                result['uid'] = chunk[2]
+                result['method'] = method
+                result = _transform_to_output_data(result, ['uid', 'method'])
+                analysis.append(result)
+        except exceptions.GenericRegressionExceptionBase as e:
+            print("info: unable to perform regression analysis on function '{0}'.".format(chunk[2]))
```

Comment:
This should rather be a warning (from perun/utils/log) so we can filter these stuff out in the future.